InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Validations.Modify.cs

```diff
@@ -0,0 +1,382 @@
+
+            invalidInternExeption.AddData(
+                key: nameof(Intern.UpdatedDate),
+                values: "Date is not recent");
+
+            var expectedInternValidException =
+                new InternValidationException(invalidInternExeption);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectInternByIdAsync(invalidIntern.Id))
+                    .ReturnsAsync(invalidIntern);
+
+            this.dateTimeBrokerMock.Setup(broker =>
+                broker.GetCurrentDateTimeOffset())
+                    .Returns(randomDateTime);
+
+            // when
+            ValueTask<Intern> modifyInternTask =
+                this.internService.ModifyInternAsync(invalidIntern);
+
+            InternValidationException actualInternValidationException =
+                await Assert.ThrowsAsync<InternValidationException>(
+                    modifyInternTask.AsTask);
+
+            // then
+            actualInternValidationException.Should().BeEquivalentTo(
+                expectedInternValidException);
+
+           this.dateTimeBrokerMock.Verify(broker =>
+                broker.GetCurrentDateTimeOffset(),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameValidationExceptionAs(
+                    expectedInternValidException))),
+                        Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+               broker.SelectInternByIdAsync(randomIntern.Id),
+                   Times.Never);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnModifyIfInternDoesntExistAndLogItAsync()
+        {
+            DateTimeOffset randomDateTimeOffset = GetRandomDateTimeOffset();
+            Intern nonExistentIntern = CreateRandomIntern();
+            nonExistentIntern.UpdatedDate = randomDateTimeOffset;
+            Intern noIntern = null;
+            var notFoundInternException = new InternNotFoundException(nonExistentIntern.Id);
+
+            var expectedInternValidationException =
+                new InternValidationException(notFoundInternException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectInternByIdAsync(nonExistentIntern.Id))
+                    .ReturnsAsync(noIntern);
+
+            this.dateTimeBrokerMock.Setup(broker =>
+                broker.GetCurrentDateTimeOffset())
+                    .Returns(randomDateTimeOffset);
+
+            // when
+            ValueTask<Intern> modifyInternTask =
+                this.internService.ModifyInternAsync(nonExistentIntern);
+
+            // then
+            await Assert.ThrowsAsync<InternValidationException>(
+               modifyInternTask.AsTask);
+
```

Comment:
It is an old approach. Assign this to variable called `actualInternValidationException` and check if it is equivalent to expected exception